In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
data = pd.read_csv(r'C:\Users\hp\Downloads\StudentsPerformance.csv')

In [5]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [6]:
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# 1.Find out how many males and females participated in the test.

In [7]:
data.groupby('gender')['gender'].count()

gender
female    518
male      482
Name: gender, dtype: int64

The Total Number of Female Participants is 518 which is 36 numbers more than the males.So females are more interested to attend the test.

# 2.What do you think about the students' parental level of education?

In [8]:
data.groupby('parental level of education')['parental level of education'].count()

parental level of education
associate's degree    222
bachelor's degree     118
high school           196
master's degree        59
some college          226
some high school      179
Name: parental level of education, dtype: int64

Majority of the parents are educated from some college and very close to it is the number of parents with associate's degree,Parents with cery hich level of qualification like master's degree (59)is only one fourth of the parents with some college level(226) education.

# 3.Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course

In [14]:
data.groupby('gender')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

In [10]:
data.groupby('test preparation course')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

# 4. What do you think about the scoring variation for math, reading and writing
based on

In [12]:
data.groupby('gender')[['math score','reading score','writing score']].var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

The variation for math score is higher for female so they are having more extreme values compared to male students.For reading score the variation is lower compared to math score and the difference between the male and the female candidates are also lower.For writing also the the female students are having larger extreme values compared to the male which is 18 less than that of female.

In [13]:
data.groupby('test preparation course')[['math score','reading score','writing score']].var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

The variation of score in all the three sections are more uniform in case of preparation course completed students that toot the difference is huge in case of writing were the students who took writing practices scored more evenly that the ones who didn't take the preparation test.

# 5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.

In [12]:
data.describe()

math score  reading score  writing score
count  1000.000000    1000.000000    1000.000000
mean     66.001000      69.195000      68.119000
std      15.569567      14.706436      15.226381
min       0.000000      17.000000      10.000000
25%      57.000000      59.000000      58.000000
50%      66.000000      70.000000      69.000000
75%      77.000000      79.250000      79.000000
max     100.000000     100.000000     100.000000

The quartile value is given by the describe command and we understand that the 3rd quartile is 77,79.25,79 for math,reading,writing respectively 

In [15]:
data.quantile(0.75)['math score']

77.0

The students who scored 77 or greater is to be given the bonus mark by the management as the top 25 percentage is the 3rd quartile which is 77

In [17]:
d1=data['math score']
d1[d1>77].count()

231

So there are 231 students to get bonus mark in the group